weldx/time.py

```diff
@@ -0,0 +1,125 @@
+            An absolute reference point in time (timestamp). The return values of all
+            accessors that return relative times will be calculated towards this
+            reference time.
+            This will turn the data of this class into absolute time values if relative
+            times are passed as ``time`` parameter. In case ``time`` already contains
+            absolute values and this parameter is set to ``None``, the first value of
+            the data will be used as reference time.
+
+        """
+        pass
+
+    def __add__(self, other):
+        # discuss what this is supposed to do. There are multiple possibilities
+        pass
+
+    def __sub__(self, other):
+        # discuss what this is supposed to do. There are multiple possibilities
+        pass
+
+    def __eq__(self, other: Union[types_time_like, "Time"]) -> bool:
+        pass
+
+    def __le__(self, other: Union[types_time_like, "Time"]) -> bool:
+        pass
+
+    def all_close(self, other: Union[types_time_like, "Time"], tolerance) -> bool:
+        """Return `True` if another object compares equal within a certain tolerance."""
+
+    def as_quantity(self) -> Quantity:
+        """Return the data as `pint.Quantity`."""
+
+    def as_time_delta_index(self) -> TimedeltaIndex:
+        """Return the data as `pandas.TimedeltaIndex`."""
+
+    def as_datetime_index(self) -> DatetimeIndex:
+        """Return the data as `pandas.DatetimeIndex`."""
+
+    def as_timestamp(self) -> Timestamp:
+        """Return the data as `pandas.Timestamp`."""
+
+    @property
+    def data_array(self) -> DataArray:
+        """Return the internally stored data."""
+
+    @data_array.setter
+    def data_array(self, data_array: DataArray):
+        """Set the internal data."""
+
+    @property
+    def reference_time(self) -> "Time":
+        """Get the reference time."""
+
+    @property
+    def reference_time_as_timestamp(self) -> Timestamp:
+        """Get the reference time as `pandas.Timestamp`."""
+        return self.reference_time.as_timestamp()
+
+    @reference_time.setter
+    def reference_time(self, time_ref: Union[types_timestamp_like, "Time"]):
+        """Set the reference time."""
+        pass
+
+    @property
+    def has_reference_time(self) -> bool:
+        """Return `True` if the class has a reference time and `False` otherwise."""
+
+    @property
+    def length(self) -> int:
+        """Return the length of the data."""
+
+    @property
+    def is_timestamp(self) -> bool:
+        """Return `True` if the data represents a timestamp and `False` otherwise."""
+        return self.length == 1 and self.has_reference_time
+
+    def max(self) -> "Time":
+        """Get the maximal time of the data."""
+
+    def min(self) -> "Time":
+        """Get the minimal time of the data."""
+
+    @staticmethod
+    def union(times=List[Union[types_time_like, "Time"]]) -> "Time":
+        """Calculate the union of multiple `Time` instances.
```

Comment:
It would be awesome to have this work as a class method and as a regular method (creating union of existing Time instance with other)

maybe this can get us there:
https://stackoverflow.com/a/28238212

Reply:
Nice idea/find. We will get it done 😉 But first I have to [learn more about descriptors](https://docs.python.org/3/howto/descriptor.html)